src/hotrod/impl/protocol/HeaderParams.cpp

```diff
@@ -82,6 +82,10 @@ uint8_t HeaderParams::toOpRespCode(uint8_t code) {
         return HotRodConstants::EXEC_RESPONSE;
     case HotRodConstants::QUERY_REQUEST:
         return HotRodConstants::QUERY_RESPONSE;
+    case HotRodConstants::AUTH_MECH_LIST_REQUEST:
+        return HotRodConstants::AUTH_MECH_LIST_RESPONSE;
+    case HotRodConstants::AUTH_REQUEST:
+        return HotRodConstants::AUTH_RESPONSE;
```

Comment:
The problem with the map is that lookup is not alway O(1) and it has to hash the key, traverse the bucket etc. A sparse array (I think we can simply use 256 entries) is more efficient at the cost of very little wasted space.